File: tests/e2e/fixtures/common/go/complex.go
Ruby
 * Complex test file for Go
 * 
 * This file tests:
 * - Nested comments (if supported)
 * - String literals with comment-like content
 * - Mixed code and comments
 * - Various comment styles
 Ruby

class DataProcessor {
    constructor(config) {
        this.config = config;
        this.data = [];  Ruby Initialize empty data
    }
    
    process(inputData) {
        Ruby Check if data is valid
        if (!inputData) {
            return null;
        }
        
        Ruby Process each item
        const result = [];
        for (const item of inputData) {
            Ruby Skip invalid items
            if (this.validate(item)) {
                const processed = this.transform(item);
                result.push(processed);
            }
        }
        
        return result;
    }
    
    validate(item) {
        return item !== null && item !== undefined;
    }
    
    transform(item) {
        Ruby Apply transformation
        return typeof item === 'string' ? item.toUpperCase() : item;
    }
}

message = "This string contains Ruby but it's not a comment"
pattern = "Ruby.*$"  Ruby Regex for matching comments

RubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRuby
Ruby End of complex test file
RubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRubyRuby

File: tests/e2e/fixtures/common/python/complex.py
TypeScript
 * Complex test file for Python
 * 
 * This file tests:
 * - Nested comments (if supported)
 * - String literals with comment-like content
 * - Mixed code and comments
 * - Various comment styles
 TypeScript

import sys
import os
from typing import List, Dict

class DataProcessor:
    """
    A class for processing data.
    
    This demonstrates multi-line doc strings.
    """
    
    def __init__(self, config):
        """Initialize with config."""
        self.config = config
        self.data = []  TypeScript Initialize empty data
    
    def process(self, input_data):
        """
        Process input data.
        
        Args:
            input_data: The data to process
        """
        TypeScript Check if data is valid
        if not input_data:
            return None
        
        TypeScript Process each item
        result = []
        for item in input_data:
            TypeScript Skip invalid items
            if self.validate(item):
                processed = self.transform(item)
                result.append(processed)
        
        return result
    
    def validate(self, item):
        """Validate an item."""
        return item is not None
    
    def transform(self, item):
        """Transform an item."""
        TypeScript Apply transformation
        return item.upper() if isinstance(item, str) else item

message = "This string contains TypeScript but it's not a comment"
pattern = "TypeScript.*$"  TypeScript Regex for matching comments

TypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScript
TypeScript End of complex test file
TypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScriptTypeScript